packages/cdktf-cli/bin/cmds/helper/stack-dependencies.ts

```diff
@@ -0,0 +1,39 @@
+import { SynthesizedStack } from "./synth-stack";
+
+export class StackDependencies {
+  public pendingStacks: SynthesizedStack[] = [];
+  public inflightStacks: SynthesizedStack[] = [];
+  public deployedStacks: SynthesizedStack[] = [];
+
+  constructor(private readonly stacks: SynthesizedStack[]) {
+    this.pendingStacks = [...this.stacks];
+  }
+
+  public startRun(stack: SynthesizedStack): void {
+    this.pendingStacks = this.pendingStacks.filter(
+      (item) => item.name !== stack.name
+    );
+    this.inflightStacks.push(stack);
+  }
+
+  public finishRun(stack: SynthesizedStack): void {
+    this.inflightStacks = this.inflightStacks.filter(
+      (item) => item.name !== stack.name
+    );
+    this.deployedStacks.push(stack);
+  }
+
+  public pendingDeployableStacks(): SynthesizedStack[] {
```

Comment:
Nit: Should we make this a getter? We don't have a common codestyle / linting for such things (yet) – no hard feelings though 😁